test/cpp/jit/test_gpu.cpp

```
@@ -11777,6 +11777,54 @@ TEST(NVFuserTest, FusionVectorization1_CUDA) {
       &fusion, cg_outputs, aten_inputs, {aten_output}, __LINE__, __FILE__);
 }
 
+TEST(NVFuserTest, FusionVectorizationMisaligned_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  auto tv1 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  auto tv2 = add(tv0, tv1);
+  fusion.addOutput(tv2);
+
+  tv2->split(-1, 64);
+  tv2->split(-1, 4);
+
+  tv2->axis(0)->parallelize(ParallelType::BIDx);
+  tv2->axis(1)->parallelize(ParallelType::TIDx);
+
+  auto c0 = tv0->cache_after();
+  //auto c2 = tv2->cache_before();
+  c0->computeAt(tv2, -1);
+
+  std::vector<TensorView*> vectorized_tvs = {c0};
+  for (auto tv : vectorized_tvs) {
+    tv->split(-1, 4);
```

I don't think this should be allowed as `c0` was inlined into `tv2` and now you're trying to transform the axis within the compute at. This should throw an error.

---

The computation needs to be inlined with the vectorized read / write with misaligned address.